en/simulation/hitl.md

```diff
@@ -1,101 +1,133 @@
+1. Select Airframe
+   1. Open **Setup > Airframes**
+   1. Select a [compatible airframe](#compatible_airframe) you want to test.
+      Generally you'll select *HILStar* for Fixed Wing/X-Plane simulator and a *HIL QuadCopter* option for copters (and jMAVSim or Gazebo).
 
-* **Make sure QGroundControl is not running**
+      ![Select Airframe](../../assets/gcs/qgc_hil_config.png)
+1. Setup UDP
+   1. Under the *General* tab of the settings menu, uncheck all *AutoConnect* boxes except for **UDP**. 
 
-* Connect the autopilot to the computer and allow it a few seconds to boot.
+      ![QGC Auto-connect settings for HITL](../../assets/gcs/qgc_hitl_autoconnect.png)
+1. (Gazebo only) Set the `SYS_COMPANION` parameter to `921600` (see [PX4 User Guide > Parameters](https://docs.px4.io/en/advanced_config/parameters.html#finding-a-parameter) for instructions on how to change parameters).
 
-* Run Gazebo in HITL mode 
-```
-gazebo /Tools/sitl_gazebo/worlds/iris.world
-```
-* Then run QGroundControl - it will auto-connect.
+1. (Optional) Configure Joystick and Failsafe.
 
-### Using jMAVSim \(Quadrotor only\)
+   Set the following [parameters](https://docs.px4.io/en/advanced_config/parameters.html#finding-a-parameter) in order to use a joystick instead of an RC remote control transmitter:
+   * [COM_RC_IN_MODE](../advanced/parameter_reference.md#COM_RC_IN_MODE) to "Joystick/No RC Checks". This allows joystick input and disables RC input checks.
+   * [NAV_DLL_ACT](../advanced/parameter_reference.md#NAV_DLL_ACT) to "Disabled". This ensures that no RC failsafe actions interfere when not running HITL with a radio control.
 
-* **Make sure QGroundControl is not running**
-* Run jMAVSim in HITL mode \(replace the serial port name `/dev/ttyACM0` if necessary - e.g. on Mac OS this would be `/dev/tty.usbmodem1`\):
+   > **Tip** The *QGroundControl User Guide* also has instructions on [Joystick](https://docs.qgroundcontrol.com/en/SetupView/Joystick.html) and [Virtual Joystick](https://docs.qgroundcontrol.com/en/SettingsView/VirtualJoystick.html) setup.
+   
+Once configuration is complete, **close** *QGroundControl* and disconnect the flight controller hardware from the computer. 
 
+### Simulator Setup
+
+Follow the appropriate setup steps for your simulator in the following sections.
+
+#### Gazebo
+
+> **Note** Make sure *QGroundControl* is not running!
+
+1. Update the environment variables: 
+  ```sh
+  cd <Firmware_clone>
+  make posix_sitl_default gazebo
+  source Tools/setup_gazebo.bash $(pwd) $(pwd)/build/posix_sitl_default
   ```
-  ./Tools/jmavsim_run.sh -q -d /dev/ttyACM0 -b 921600 -r 250
+1. Open the vehicle model's sdf file (e.g. **Tools/sitl_gazebo/models/iris/iris.sdf**).
+1. Under the `mavlink_interface plugin` section, change the `serialEnabled` and `hil_mode` parameters to `true`. 
+
+  ![HIL Parameters](../../assets/simulation/gazebo_sdf_model_hil_params.png)
+1. Replace the `serialDevice` parameter (`/dev/ttyACM0`) if necessary.
```

Comment:
@arwagoner How will I know if changing serialDevice is necessary? Is this also set in the SDF file? Is it obvious - ie can I just search for it, or is the section name useful?

Reply:
If the user is connecting the autopilot to the computer using a serial link, the serial device will likely be /dev/ttyACM0. If the autopilot is connected via TELEM2, then the device might change to /dev/ttyUSB0. If there are multiple devices connected, the device might show up as /dev/ttyUSB1 or /dev/ttyACM1. An easy way to check on Ubuntu is to plug in the autopilot, open up a terminal, and type `dmesg | grep "tty"`. The correct device will be the last one shown.

Reply:
Thanks. I will add a note for that.